x-pack/plugins/ml/server/models/job_service/jobs.ts

```diff
@@ -257,6 +259,32 @@ export function jobsProvider(client: IScopedClusterClient, mlClient: MlClient) {
     return { jobs, jobsMap };
   }
 
+  async function getJobForCloning(jobId: string) {
+    const datafeedId = await getDatafeedIdByJobId(jobId);
+
+    const [{ body: jobResults }, { body: datafeedResults }] = await Promise.all([
+      mlClient.getJobs<MlJobsResponse>({ job_id: jobId, exclude_generated: true }),
+      mlClient.getDatafeeds<MlDatafeedsResponse>({
+        datafeed_id: datafeedId,
+        exclude_generated: true,
+      }),
+    ]);
+
+    let datafeed;
+    if (Array.isArray(datafeedResults?.datafeeds) && datafeedResults.datafeeds.length === 1) {
+      datafeed = datafeedResults?.datafeeds[0];
+    }
+
+    // create jobs objects containing job stats, datafeeds, datafeed stats and calendars
```

Comment:
looks like this comment has been copied from somewhere else.
the check below to find the job could be the same as the datafeed check above.

Reply:
Updated here https://github.com/elastic/kibana/pull/88898/commits/6ea550868c19746c3fe817c19195cbb492f36eba